### DISCUSSION TITLE
Indexer Unregister

### DISCUSSION THREAD

**Discussion by mzxyz** [subquery/subql#1044] [Frontier TestNet - Bug Reporting]
Please add the issues which related with `unregister indexer` in this discussion.

**To unregister from the network:**

- Please make sure you have change network status of all your history projects to `NOT INDEXING` with `announce not indexing`
- Projects with ongoing service agreements can not announce not indexing, for most cases indexer won't got this issue as the agreement in Testnet only have 1 day validity period.
- After trigger `unregister` action and the transaction processed, the indexer will be unregistered from the network and all the accounts and projects data will be remove from the coordinator service.

**Comment by mzxyz** [comment]
QA:

Q: Failed to call unregister and get transaction revert with `Can not unregister from the network due to running indexing projects`
A: There has history projects removed from you coordinator service but still with the INDEXING or READY status on network.  Try to add these projects to check the status, and change the status to `NOT INDEXING`:

```
- QmTLPAeVhNNxU69NhcsKHRrCXBhmQYykqkszwETRS75A76
- QmTQTnBTcvv3Eb3M6neDiwuubWVDAoqyAgKmXtTtJKAHoH
- QmVcksANdHhnA2YEguJ8LvsftNkipTwkgV9eJyDPZFUqHm
```

**Comment by kw1knode** [comment]
Currently unable to unregister. 

```transaction failed (transactionHash="0x5aa2871360ae3bb59c5417dc2d9981e898ad1e60a6dda13cdb04b0b07021c31e", transaction={"hash":"0x5aa2871360ae3bb59c5417dc2d9981e898ad1e60a6dda13cdb04b0b07021c31e","type":2,"accessList":null,"blockHash":null,"blockNumber":null,"transactionIndex":null,"confirmations":0,"from":"0x6309055061be02451232957DA4aA66e65fC36ca8","gasPrice":{"type":"BigNumber","hex":"0xa0eebb00"},"maxPriorityFeePerGas":{"type":"BigNumber","hex":"0x59682f00"},"maxFeePerGas":{"type":"BigNumber","hex":"0xa0eebb00"},"gasLimit":{"type":"BigNumber","hex":"0x0f4240"},"to":"0xe5B4EA276e1030F55790bB8695bC1f8362241bb9","value":{"type":"BigNumber","hex":"0x00"},"nonce":72,"data":"0x612151cb","r":"0x7836ba4658f0599df6a2065a5a4879cec842d704a872e8d52593dc77f5343593","s":"0x44a984425da8d709f811256690b68993340b8b5181ef270953f23d40b3437c81","v":1,"creates":null,"chainId":0}, receipt={"to":"0xe5B4EA276e1030F55790bB8695bC1f8362241bb9","from":"0x6309055061be02451232957DA4aA66e65fC36ca8","contractAddress":null,"transactionIndex":0,"gasUsed":{"type":"BigNumber","hex":"0xcc26"},"logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","blockHash":"0x1c588d3fe73c8ffac876129349e54cc65ec0ad3ed5472ad73de47fec475cfdae","transactionHash":"0x5aa2871360ae3bb59c5417dc2d9981e898ad1e60a6dda13cdb04b0b07021c31e","logs":[],"blockNumber":325122,"confirmations":1,"cumulativeGasUsed":{"type":"BigNumber","hex":"0xcc26"},"effective```

**Reply by mzxyz** [reply]
Have you check all the list projects: https://github.com/subquery/subql/discussions/1044#discussioncomment-2787191 are with not indexing status?

**Comment by zhanglonglongSH** [comment]
tip： The controller account has insufficient funds to pay the transaction fee. Please top up your controller account ASAP. 
But I have 1.9SQN and more than 2000 SQT in my wallet.

**Reply by zhanglonglongSH** [reply]
transaction failed (transactionHash="0x70c93968fd2dc0e4381126c22c09f9598cddcecdd55b83677e92134dcc6ebd19", transaction={"hash":"0x70c93968fd2dc0e4381126c22c09f9598cddcecdd55b83677e92134dcc6ebd19","type":2,"accessList":null,"blockHash":null,"blockNumber":null,"transactionIndex":null,"confirmations":0,"from":"0x1bf15607789501E246d999ae7AE423c6DAa0A532","gasPrice":{"type":"BigNumber","hex":"0xa0eebb00"},"maxPriorityFeePerGas":{"type":"BigNumber","hex":"0x59682f00"},"maxFeePerGas":{"type":"BigNumber","hex":"0xa0eebb00"},"gasLimit":{"type":"BigNumber","hex":"0x0f4240"},"to":"0xe5B4EA276e1030F55790bB8695bC1f8362241bb9","value":{"type":"BigNumber","hex":"0x00"},"nonce":132,"data":"0x612151cb","r":"0xa96429062331ccea30f1f4ffc164974a2fe80d298b954c5e876ccbb4fd2f23da","s":"0x1c6e0b48142a97d3b0b9636184b0a377530873a5b3636c5d3c972063d3d9cb86","v":1,"creates":null,"chainId":0}, receipt={"to":"0xe5B4EA276e103 ....

**Reply by mzxyz** [reply]
The balance refer to the DEV value display in controller account section, not SQT or SQN

**Reply by zhanglonglongSH** [reply]
<img width="1124" alt="image" src="https://user-images.githubusercontent.com/17761963/169683525-436a6a2a-75bd-4800-a436-f0646f1b035b.png">
I have enough tokens in my wallet

**Reply by zhanglonglongSH** [reply]
it's OK now ， Thanks

**Comment by igoodvvin** [comment]
When unregistration, transaction is not executed, details are below.

Indexer Account Balance: 0.4886 DEV
Controller Account Balance: 0.2805 DEV

Unregister Indexer Account
`transaction failed (transactionHash="0xa471ba8cc39cc4872e2b752f858078f0922241096ae11f1aa351608067613169", transaction={"hash":"0xa471ba8cc39cc4872e2b752f858078f0922241096ae11f1aa351608067613169","type":2,"accessList":null,"blockHash":null,"blockNumber":null,"transactionIndex":null,"confirmations":0,"from":"0x1bBF344e284043efC3Be114A1b7320D11A3d80Fc","gasPrice":{"type":"BigNumber","hex":"0xa0eebb00"},"maxPriorityFeePerGas":{"type":"BigNumber","hex":"0x59682f00"},"maxFeePerGas":{"type":"BigNumber","hex":"0xa0eebb00"},"gasLimit":{"type":"BigNumber","hex":"0x0f4240"},"to":"0xe5B4EA276e1030F55790bB8695bC1f8362241bb9","value":{"type":"BigNumber","hex":"0x00"},"nonce":31,"data":"0x612151cb","r":"0xb3153cec7bfacda702fb85efc40b629a57b8a165a07b51ca5646e74e03e5bf2f","s":"0x5d2b6f3ee6ccc3391347248fca82c6ec966da1aea70ee3eba6df6e0a9871382c","v":1,"creates":null,"chainId":0}, receipt={"to":"0xe5B4EA276e1030F55790bB8695bC1f8362241bb9","from":"0x1bBF344e284043efC3Be114A1b7320D11A3d80Fc","contractAddress":null,"transactionIndex":1,"gasUsed":{"type":"BigNumber","hex":"0xcc26"},"logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","blockHash":"0x83f3c28873531f6b67416bfb657f1e958b1bf934eba0bbe5575d3d89f6463c04","transactionHash":"0xa471ba8cc39cc4872e2b752f858078f0922241096ae11f1aa351608067613169","logs":[],"blockNumber":329946,"confirmations":1,"cumulativeGasUsed":{"type":"BigNumber","hex":"0x01d56f"},"effectiveGasPrice":{"type":"BigNumber","hex":"0x9502f900"},"status":0,"type":0,"byzantium":true}, code=CALL_EXCEPTION, version=providers/5.5.1)`

**Reply by igoodvvin** [reply]
In the UI, indexing of all projects was disabled, but the explorer displayed one project in offline status, which was the reason for the above error when unregistering the indexer. To avoid the error, you need to add the project that you see in the explorer by deployment id and disable indexing. In the end, everything worked out.
